via/static/scripts/video_player/hooks/use-app-layout.ts

```diff
@@ -0,0 +1,53 @@
+import type { RefObject } from 'preact';
+import { useCallback, useLayoutEffect, useState } from 'preact/hooks';
+
+type AppSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl';
+
+/**
+ * Watch for resizes on `element` and invoke `onSizeChanged` callback
+ */
+export function observeElementSize(
+  element: Element,
+  onSizeChanged: (width: number, height: number) => void
+): () => void {
+  const observer = new ResizeObserver(() =>
+    onSizeChanged(element.clientWidth, element.clientHeight)
+  );
+  observer.observe(element);
+  return () => observer.disconnect();
+}
+
+/**
+ * Return a string representing the relative size of `appContainer`
+ */
+export function useAppLayout(appContainer: RefObject<HTMLDivElement | null>) {
```

Comment:
The API design of this hook currently doesn't allow for reacting to changes in the ref's value during the lifetime of the component. I think that's likely an acceptable limitation, as the hook doesn't pretend to be generic, but we should be clear about it.

Reply:
Can you help me understand the implications of this a little better? The internal hooks have `appContainer` as a dependency, and this general setup reflects other recent hooks that take refs as an argument/option. Can you provide a scenario (not necessarily one we'd run into with this project, but in general) where this could pose an issue?